util/OptionsDB.h

```diff
@@ -177,10 +177,19 @@ class FO_COMMON_API OptionsDB
     void        Add(const std::string& name, const std::string& description, T default_value,
                     const ValidatorBase& validator = Validator<T>(), bool storable = true)
     {
-        if (m_options.find(name) != m_options.end())
-            throw std::runtime_error("OptionsDB::Add<>() : Option " + name + " was specified twice.");
-        m_options[name] = Option(static_cast<char>(0), name, default_value, default_value,
-                                 description, validator.Clone(), storable, false);
+        std::map<std::string, Option>::iterator it = m_options.find(name);
+        boost::any value = default_value;
+        if (it != m_options.end()) {
+            if (it->second.recognized)
+                throw std::runtime_error("OptionsDB::Add<>() : Option " + name + " was specified twice.");
+            try {
+                // This option was previously specified externally but was not recognized at the time, attempt to parse the value found there
+                value = validator.Validate(it->second.ValueToString());
+            } catch (boost::bad_lexical_cast&) {
+                // TODO: log error, don't need to throw because this is something that could be caused by the user messing with the config file, and we have a sensible default to fall back on (?)
```

Comment:
Ah yeah, of course.  I'll do that in my remember-window-positions branch too then. (now that I actually look at Logger.h, it's only 38 lines anyway)